src/enkf/gridio_fv3reg.f90

```diff
@@ -1064,6 +1146,58 @@ subroutine writegriddata_pnc(vars3d,vars2d,n3d,n2d,levels,ndim,grdin,no_inflate_
     real(r_single), dimension(npts,ndim,nbackgrounds,1), intent(inout) :: grdin
     logical, intent(in) :: no_inflate_flag
   end subroutine writegriddata_pnc
+  subroutine type_bound_setupfile(this,fileid1,fv3fn1,fileid2,fv3fn2)
+       class (type_fv3lamfile) :: this  
+       integer(i_kind) fileid1
+       integer(i_kind), optional :: fileid2
+       character(len=*)::fv3fn1
+       character(len=*),optional ::fv3fn2
+       if (present (fileid2)) then
+         this%l_filecombined=.false. 
+         this%fv3lamfilename(1)=trim(fv3fn1)
+         this%fv3lamfilename(2)=trim(fv3fn2)
+         this%fv3lam_fileid(1)=fileid1
+         this%fv3lam_fileid(2)=fileid2
+       else
+         this%l_filecombined=.true. 
+         this%fv3lamfilename(1)=fv3fn1
+         this%fv3lam_fileid(1)=fileid1
+       endif
+   end subroutine type_bound_setupfile
+   subroutine type_bound_getidfn(this,vnamloc,fileid,fv3fn)
+   class (type_fv3lamfile) :: this  
+   integer(i_kind) fileid
+   character(len=*)::fv3fn,vnamloc
+   if (.not.this%l_filecombined) then
+     if(ifindstrloc(vardynvars,vnamloc)> 0)  then  
+       fv3fn=trim(this%fv3lamfilename(1))
+       fileid=this%fv3lam_fileid(1)
+     else if(ifindstrloc(vartracers,vnamloc)> 0)  then  
+       fv3fn=trim(this%fv3lamfilename(2))
+       fileid=this%fv3lam_fileid(2)
+     else
+       write(6,*)"the varname ",trim(vnamloc)," is not recognized in the ype_bound_getidfn, stop"
+       call stop2(23)
+     endif
+   else
+     fv3fn=trim(this%fv3lamfilename(1))
+     fileid=this%fv3lam_fileid(1)
+   endif
+   end subroutine type_bound_getidfn
+   function ifindstrloc(str_array,strin)
+     integer(i_kind) ifindstrloc
+     character(len=max_varname_length),dimension(:) :: str_array
+     character(len=*) :: strin
+     integer(i_kind) i
+     ifindstrloc=0
+     do i=1,size(str_array)
+      if(trim(str_array(i)) == trim(strin)) then
+        ifindstrloc=i
+        exit
+      endif
+     enddo
```

Comment:
Please ensure at least two space indentation for if statements and do loops.

Reply:
@MichaelLueken-NOAA When I finished the steps (till git push .. ), I have an error message when doing the cmake:
^^

Found BUFR library BUFR_LIBRARY-NOTFOUND
Could not find BUFR library, so building from libsrc
searching for source for bufr_v10.2.5 in
didn't find directory
WARNING: Did not find 10.2.5 of bufr, looking for alternates
CMake Error at cmake/Modules/FindBUFR.cmake:44 (add_subdirectory):
  add_subdirectory given source
  "/scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-gsi-pr/GSI/libsrc/bufr" which is not
  an existing directory.
Call Stack (most recent call first):
  CMakeLists.txt:223 (find_package)

find: ‘/sigio’: No such file or directory
find: ‘/sorc’: No such file or directory
V
And, then  the cmake would become idle there,   Would you let me know what I should do? 
Thanks.

Reply:
I found it was related to the recent change of modules on hera.  I will see if I can have it work.